### PR TITLE
ability to add script plugins to oref0-pump-loop

### PR DESCRIPTION
**Pull request by efidoman**
This PR adds the ability to inject custom script(s) to run synchronously at the end of every oref0-pump-loop. Any script files with executable flag set will be run as follows:

If the script file is placed in the plugins/every folder then the script runs at the end of every loop.
If the script file is placed in the plugins/once folder then the script once at the end of the current loop.

Script files placed in the plugins/once folder will be removed once run. This allows for a use case such as the following:

Logger and Lookout both call a pump utility called fakemeter that helps out users in offline mode by sending the current CGM (Dexcom) glucose value as a meter glucose value. This shows the glucose on the pump every 5 minutes and allows you to quickly see the current glucose value on the pump. Since this script requires communication with the pump, it will collide with oref0 pump communications on occasion. This oref0-pump-loop plugin capability will allow Logger and Lookout to run fakemeter synchronously at the end of the pump loop and avoid any radio collisions.

### CODE REVIEW

File: bin/oref0-pump-loop.sh [hunk 1]
```diff
@@ -95,6 +96,38 @@ main() {
...
+
+function run_plugins {
+        once=plugins/once
+        every=plugins/every
+        mkdir -p $once
+        mkdir -p $every
+        echo "scripts placed in this directory will run once after curent loop and be removed" > $once/readme.txt
+        echo "scripts placed in this directory will run after every loop" > $every/readme.txt
+        find $once/* | while read file; do run_script "$file" -d ; done
+        find $every/* | while read file; do run_script "$file" ; done
```
Comment (scottleibrand) [inline bin/oref0-pump-loop.sh]: These will attempt to run the readme.txt as a script.  We should probably limit the find to just `*.sh` files.

Comment (efidoman) [reply bin/oref0-pump-loop.sh]: The run_script function checks for -x (executable flag) so it should skip any others. It has skipped all others in my testing so far.

Comment (scottleibrand) [reply bin/oref0-pump-loop.sh]: Ah, that works.  Added a comment noting that's how it works.

**Comment by scottleibrand** [comment]
Code looks good.  If someone can post test results, I think we can merge this.

**Comment by efidoman** [comment]
needs a quick review again. I found an issue while testing where the plugins didn't run upon any loop failure. The intention is for plugins to run at the end of every pump loop regardless of pass or fail. Also, I updated the comments and logging a bit.

**Comment by efidoman** [comment]
Here's my latest test results that include the Logger PR to leverage this new plugin capability for running fakemeter:

```
o bolus needed. Settings less than 15 minutes old. Your instance of oref0 [0.7.0-dev, dev] is out-of-date by 28 commits: you may want to consider updating.
You are currently running a development branch of oref0.  Such branches change frequently.
Please read the latest PR notes and update with the latest commits to dev before reporting any issues.
Completed oref0-pump-loop at Wed Feb 27 08:28:30 EST 2019

Running plugin script (plugins/once/run_fakemeter.sh)... 
2019/02/27 08:28:31 connected to CC111x radio on /dev/spidev5.1
2019/02/27 08:28:31 setting frequency to 916.650
2019/02/27 08:28:31 model 522 pump
2019/02/27 08:28:34 disconnecting CC111x radio on /dev/spidev5.1
Completed plugin script (plugins/once/run_fakemeter.sh). 
Running plugin script (plugins/every/plugins.sh)... 
Plugins Starting - Wed Feb 27 08:28:34 EST 2019
Completed plugin script (plugins/every/plugins.sh). 
Running plugin script (plugins/every/yo.sh)... 
/root/myopenaps
Completed plugin script (plugins/every/yo.sh). 

Starting oref0-pump-loop at Wed Feb 27 08:29:03 EST 2019 with 1 second wait_for_silence:
```

File: bin/oref0-pump-loop.sh [hunk 2]
```diff
@@ -92,9 +92,42 @@ main() {
...
```
Comment (jpcunningh) [inline bin/oref0-pump-loop.sh]: I would recommend adding `-executable` to the find commands to eliminate the need for the `if` block in run_plugins testing for execution permissions.  It makes run_plugins a little cleaner.
```
find $once/* -executable | while read file; do run_script "$file" -d ; done
find $every/* -executable | while read file; do run_script "$file" ; done
```

File: bin/oref0-pump-loop.sh [hunk 3]
```diff
@@ -87,32 +87,27 @@ main() {
             touch /tmp/pump_loop_success
             echo Completed oref0-pump-loop at $(date)
             update_display
-            run_plugins
             echo
         else
             # pump-loop errored out for some reason
             fail "$@"
         fi
+        run_plugins
```
Comment (scottleibrand) [inline bin/oref0-pump-loop.sh]: Moving it here won't help make it run on failures, as the fail function calls exit 1.  You'll instead need to add run_plugins after update_display there as well as above.

Comment (efidoman) [reply bin/oref0-pump-loop.sh]: Thank you. I was not aware. I have updated run_plugins to run in both places so it runs either way.

Comment (scottleibrand) [reply bin/oref0-pump-loop.sh]: I just tweaked the placement to be more consistent between the two cases.  LMK if you don't think that will do the right thing.

Comment (efidoman) [reply bin/oref0-pump-loop.sh]: LGTM

**Comment by efidoman** [comment]
It's ready. I've been watching the logs and testing - works great.

**Pull request merged**